addons/Hierarchical_Table_Of_Contents/test/KeyboardNavigationTests.js

```diff
@@ -0,0 +1,244 @@
+            $tableBody.append(this.generateChapterWithID(4));
+
+            return $table;
+    },
+
+    generateTableHeader: function () {
+            let $header = $(`
+                <tr>
+                    <td>Header</td>
+                </tr>
+            `);
+            $header.addClass("hier_report-header");
+
+            return $header;
+    },
+
+    generateChapterWithID: function (id) {
+            let $tr = $(`
+                <tr class="treegrid-${id} treegrid-collapsed hier_report-chapter">
+                    <td>
+                        <div class="treegrid-expander treegrid-expander-collapsed"></div>
+                        <div class="text-wrapper">Chapter</div>
+                    </td>
+                </tr>
+            `);
+
+            return $tr;
+    },
+
+    generateTableRowWithIdAndParent: function (id, parentId) {
+            let $tr = $(`
+                <tr class="treegrid-${id} treegrid-parent-${parentId}" style="display: none">
+                    <td>
+                        <div class="text-wrapper">Page with parent</div>
+                    </td>
+                </tr>
+            `);
+
+            return $tr;
+    },
+
+    generateTableRowWithId: function (id) {
+            let $tr = $(`
+                <tr class="treegrid-${id}">
+                    <td>
+                        <div class="text-wrapper">
+                            Page
+                        </div>
+                    </td>
+                </tr>
+            `);
+
+            return $tr;
+    },
+
+    getTreeGrid: function (id) {
+        return this.presenter.$view.find(`.treegrid-${id}`);
+    },
+
+    setCollapsedChapterAsCurrentElement: function () {
+        KeyboardController.prototype.switchElement.call(this.presenter.keyboardControllerObject, 1);
+    },
+
+    setLastKeyNavElementAsCurrentElement: function () {
+        KeyboardController.prototype.switchElement.call(this.presenter.keyboardControllerObject, 1);
+    },
+
+    setTableRowAfterCollapsedChapterAsCurrentElement: function () {
+        KeyboardController.prototype.switchElement.call(this.presenter.keyboardControllerObject, 3);
+    },
+
+    activateTTSWithoutReading: function() {
+        this.presenter.setWCAGStatus(true);
+        this.presenter.isFirstEnter = false;
+        this.presenter.keyboardControllerObject.keyboardNavigationActive = true;
+        this.presenter.keyboardControllerObject.setElements(this.presenter.getElementsForTTS());
+    },
+
+    'test given view when entering for the first time by keyboard navigation should mark first hyper link' : function() {
+        activateEnterEvent(this.presenter);
+
+        assertTrue(hasKeyboardNavigationActiveElementClass(this.presenter.$view.find(".treegrid-1")));
+    },
+
+    'test given view with keyNav when next element is collapsed then skip to next one' : function() {
+        activateEnterEvent(this.presenter);
```

Comment:
Jeżeli jest taka możliwość staraj się pisać testy tak, aby odpowiadały 3 krokom: GIVEN, WHEN, THEN a spacje je oddzielały:
```suggestion
        activateEnterEvent(this.presenter);
        this.setCollapsedChapterAsCurrentElement();

        activateTabEvent(this.presenter);

        assertTrue(hasKeyboardNavigationActiveElementClass(this.presenter.$view.find(".treegrid-4")));
```
Dla ułatwienia:
```suggestion
        \\ GIVEN
        activateEnterEvent(this.presenter);
        this.setCollapsedChapterAsCurrentElement();

        \\ WHEN
        activateTabEvent(this.presenter);

        \\ THEN
        assertTrue(hasKeyboardNavigationActiveElementClass(this.presenter.$view.find(".treegrid-4")));
```